larndsim/detsim.py

```diff
@@ -300,35 +317,30 @@ def get_track_pixel_map(track_pixel_map, unique_pix, pixels):
                     track_pixel_map[index][imap] = itrk
 
 @cuda.jit
-def get_track_pixel_map2(track_pixel_map, unique_pix, pixels, distances, max_distance):
+def get_track_pixel_map2(track_pixel_map, unique_pix, pixels, distances):
     """
     This kernel fills a 2D array which contains, for each unique pixel,
     an array with the track indeces associated to that pixel.
-
-    Args:
-        track_pixel_map_col (:obj:`numpy.ndarray`): 2D array that will contain the
-            association between the unique pixels array and the track indeces
-        unique_pix (:obj:`numpy.ndarray`): 1D array containing the unique pixels
-        pixels (:obj:`numpy.ndarray`): 2D array containing the pixels for each
-            track.
+    Summary of the different get_track_pixel_map
+    get_track_pixel_map, fills track_pixel_map without distance ranking
+    get_track_pixel_map3, fills track_pixel_map ranked by distances of unit pixel pitch
     """
     # index of unique_pix array
     index = cuda.grid(1)
     if index >= unique_pix.shape[0]:
         return
     upix = unique_pix[index]
 
-    for target_dist in range(max_distance):
-    
+    for target_dist in detector.NEIGHBORING_PIX_DIST:
```

Comment:
The link from track to pixels is subject to `MAX_RADIUS`.

The link from pixel to tracks is subject to `NEIGHBORING_PIX_DIST`.  It may be error-prone to have two parameters. Enlarging `NEIGHBORING_PIX_DIST` may not work when it is larger than `MAX_RADIUS` because the map from pixel to tracks require information from the link from track to pixels.

Reply:
There might be a misunderstanding here.  By definition, `NEIGHBORING_PIX_DIST` is associated with `MAX_RADIUS` (see [lines](https://github.com/DUNE/larnd-sim/blob/75ba73f5879f8f48fa8aeabed06fd69583e8f7ee/larndsim/consts/detector.py#L410-L415)). In addition, `NEIGHBORING_PIX_DIST` is not a single value, but a list of possible distances covered by the range of the response LUT in a unit of pixel pitch. Let me know if I misunderstood your comment.

Reply:
Thanks! I misunderstood the calculation of `NEIGHBORING_PIX_DIST`.